examples/oft-evm-move-adapters/sources/oft_implementation/move_oft_adapter.move

```diff
@@ -266,6 +279,38 @@ module oft::move_oft_adapter {
         oft_limit::new_oft_limit(0, oft_impl_config::rate_limit_capacity(eid))
     }
 
+    // ==================================================== Pauser Logic ===================================================
+
+    #[view]
+    public fun is_paused(): bool acquires PauserStore {
+        let admin_addr = get_admin();
+        if (!exists<PauserStore>(admin_addr)) { return false };
+        borrow_global<PauserStore>(admin_addr).paused
+    }
+
+    /// Set the pauser address (admin-only). Creates the store at current admin address if missing.
+    public entry fun set_pauser(admin: &signer, pauser: address) acquires PauserStore {
+        let admin_addr = address_of(admin);
+        assert_admin(admin_addr);
+        if (exists<PauserStore>(admin_addr)) {
+            borrow_global_mut<PauserStore>(admin_addr).pauser = pauser;
+        } else {
+            move_to<PauserStore>(admin, PauserStore { pauser, paused: false });
+        }
+    }
+
+    /// Toggle pause (pauser or admin can call)
+    public entry fun set_paused(caller: &signer, paused: bool) acquires PauserStore {
+        let admin_addr = get_admin();
+        assert!(exists<PauserStore>(admin_addr), ENOT_INITIALIZED);
+        let caller_addr = address_of(caller);
+        let store_mut = borrow_global_mut<PauserStore>(admin_addr);
+        assert!(caller_addr == admin_addr || caller_addr == store_mut.pauser, EUNAUTHORIZED);
```

Comment:
Okay, sounds good, just seems like pausing will only happen if a pauser address is set, correct? not an issue, but just a reminder that we need to set it asap